### PR TITLE
Allow target-specific parameter binding

### PR DESCRIPTION
**Pull request by hhorii**
This PR extends existing API to bind parameters by adding `EmitAction` into arguments of `getBindArgumentsImplementationFactory()`. and an appropriate `BindArgumentsImplementationFactory` can be selected based on `emit` value. For example, in case of `qem`, `BindArgumentsImplementationFactory` will generate an instance that binds parameters assuming its base file is formatted with `qem` (Zip). In case of `qe-qem`, the instance will assume target-specific format.

### CODE REVIEW

File: lib/Arguments/Signature.cpp [hunk 1]
```diff
@@ -69,23 +69,6 @@ void Signature::dump() {
   }
 }
 
-std::string Signature::serialize() {
```
Comment (taalexander) [inline lib/Arguments/Signature.cpp:72]: Why did this get moved to the header?

**Review by taalexander — DISMISSED** [review]
LGTM

File: lib/Arguments/Signature.cpp (hunk 1 above)
Comment (hhorii) [reply lib/Arguments/Signature.cpp:72]: I encountered a linker error. Probably, I need to add a dependency to link `Signature::serialize()`, but I was not able to find what library should be listed.

Comment (bcdonovan) [reply lib/Arguments/Signature.cpp:72]: Should be: QSSCArguments
https://github.com/openqasm/qe-compiler/blob/f62b76dfdf5a839998f3047abb6d1be623608932/lib/Arguments/CMakeLists.txt#L13

**Comment by hhorii** [comment]
@bcdonovan Thank you for your suggestion. I added `QSSCArguments` and `QSSCError` int dependencies of `QSSCPayload` and then successfully built codes.
(BTW, These dependencies were necessary in gcc@Linux env but not in Clang@Mac.)

**Review by taalexander — APPROVED** [review]
LGTM

**Pull request merged**